### PR TITLE
[custom-set] Add test for deduplication

### PR DESCRIPTION
**Pull request by senekor**
We should have a test case where a set is instantiated with a slice containing duplicate values.

motivated by:
https://github.com/exercism/rust/issues/1275

### CODE REVIEW

**Comment by neenjaw** [comment]
Is a set with duplicate elements a valid set?

**Comment by senekor** [comment]
sorry for opening this PR without a forum post, discussion should continue [here](https://forum.exercism.org/t/custom-set-add-test-for-deduplication/7314/2)

**Review by ErikSchierboom — APPROVED** [review]

File: exercises/custom-set/canonical-data.json [hunk 1]
```diff
@@ -131,6 +131,26 @@
             "set2": [4, 1, 3]
           },
           "expected": false
+        },
+        {
+          "uuid": "0b95204d-93fa-4b60-9499-1e34f228c336",
+          "description": "set is a subset of set with duplicates",
```
Comment (ErikSchierboom) [inline exercises/custom-set/canonical-data.json:137]: A set with with duplicates isn't technically a set 🤷

Comment (senekor) [reply exercises/custom-set/canonical-data.json:137]: Yes :sweat_smile: but it's shorter than "a set constructed from an array with duplicates" :joy:

Comment (ErikSchierboom) [reply exercises/custom-set/canonical-data.json:137]: It is, but I actually like the second version better!

Comment (senekor) [reply exercises/custom-set/canonical-data.json]: OK, [chanded it](https://github.com/exercism/problem-specifications/pull/2324/commits/915bf818ab226d0e16c7452fd6981e8684f7fb08)!

**Review by iHiD — APPROVED** [review]

**Review by IsaacG — APPROVED** [review]

**Pull request merged**